test/framework/src/main/java/org/elasticsearch/repositories/blobstore/ESBlobStoreRepositoryIntegTestCase.java

```diff
@@ -47,42 +50,50 @@
 import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertHitCount;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.instanceOf;
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.notNullValue;
 import static org.hamcrest.Matchers.nullValue;
 
 /**
- * Basic integration tests for blob-based repository validation.
+ * Integration tests for {@link BlobStoreRepository} implementations.
  */
 public abstract class ESBlobStoreRepositoryIntegTestCase extends ESIntegTestCase {
 
-    protected abstract void createTestRepository(String name, boolean verify);
-
-    protected void afterCreationCheck(Repository repository) {
+    protected abstract String repositoryType();
 
+    protected Settings repositorySettings() {
+        final Settings.Builder settings = Settings.builder();
+        settings.put("compress", randomBoolean());
+        if (randomBoolean()) {
+            long size = 1 << randomIntBetween(7, 10);
```

Comment:
This is too large :) GCS only allows up to 100Mb

Reply:
Yeah, I just saw the test failure :)